### PR TITLE
Prototype lint GitHub Action

### PR DESCRIPTION
**Pull request by namoscato**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  app:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: app
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Read .nvmrc
```
Comment (namoscato) [inline .github/workflows/lint.yml:12]: By the way, per [Specifying a Node Version](https://render.com/docs/node-version):

> By default, Render uses the latest LTS version of Node.

which is 12.16.3 at the moment. There are different ways to configure that, none of which are `.nvmrc`, but we should probably try to keep them in sync?

File: app/package.json [hunk 2]
```diff
@@ -31,7 +31,8 @@
     "start": "react-scripts start",
     "build": "react-scripts build",
     "eject": "react-scripts eject",
-    "gql-gen": "graphql-codegen --config codegen.js --require dotenv/config"
+    "gql-gen": "graphql-codegen --config codegen.js --require dotenv/config",
+    "lint": "eslint --ext ts,tsx src && prettier --check \"src/{**/*,*}.{js,ts,jsx,tsx}\""
```
Comment (namoscato) [inline app/package.json:35]: `prettier` is failing in its current state, which we can fix assuming we want to use it?

We could also split these up into separate scripts / jobs for clarity?

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        run: echo ::set-output name=NVMRC::$(cat .nvmrc)
+        id: nvm
+      - name: Use Node.js ${{ steps.nvm.outputs.NVMRC }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: '${{ steps.nvm.outputs.NVMRC }}'
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Lint TypeScript
+        run: yarn run lint
```
Comment (avimoondra) [inline .github/workflows/lint.yml:22]:
```suggestion
      - name: Lint Typescript/Javascript
        run: yarn run lint
      - name Typecheck
        run: yarn run tsc
```

another check we can add

**Comment by avimoondra** [comment]
Also, removing CircleCI ... ✌️

**Comment by wtachau** [comment]
wow, github actions -- they're taking over the world!

File: .github/workflows/lint.yml (hunk 3 above)
Comment (wtachau) [reply .github/workflows/lint.yml:22]: i think you need a `:` after name on line 23

File: .github/workflows/lint.yml (hunk 1 above)
Comment (namoscato) [reply .github/workflows/lint.yml:12]: I'm cool just leaving it for now 🤷‍♂️

File: app/package.json [hunk 4]
```diff
@@ -31,7 +31,8 @@
     "start": "react-scripts start",
     "build": "react-scripts build",
     "eject": "react-scripts eject",
-    "gql-gen": "graphql-codegen --config codegen.js --require dotenv/config"
+    "gql-gen": "graphql-codegen --config codegen.js --require dotenv/config",
+    "lint": "eslint --ext ts,tsx src && tsc --noEmit && prettier --check \"src/{**/*,*}.{js,ts,jsx,tsx}\""
```
Comment (namoscato) [inline app/package.json:35]: I was struggling to determine how to exclude `src/generated` from this glob pattern which is the reason `prettier` is still failing.

File: .github/workflows/lint.yml (hunk 3 above)
Comment (namoscato) [reply .github/workflows/lint.yml]: I added `tsc --noEmit` to the npm `lint` script to keep things in one place. Thanks for the feedback!

**Comment by avimoondra** [comment]
[Stop checking in generated files](https://trello.com/c/eQI8i9qK/91-stop-checking-in-generated-files)

**Pull request merged**